### PR TITLE
Add preserve argument to masker

### PR DESCRIPTION
**Pull request by zkamvar**
The preserve argument will allow the user to specify if the original mask should be preserved or discarded. If it's preserved, then all values that are FALSE in the original mask cannot be overwritten to TRUE.

### CODE REVIEW

**Comment by knausb** [comment]
Thanks Zhian!

**Comment by zkamvar** [comment]
No problem. 

Additionally, I was thinking it might be a good idea to include a mask method to chromR objects so that users can more easily update the mask. This way, you could impose error checking to make sure that the user doesn't do something silly like add a mask that's too short, but still a factor of the total length, or adding a numeric or character value. If the mask is optional, just simply throw an error when it doesn't exist.

```r
dp <- extract.gt(vcf, element = "DP", as.numeric = TRUE)
my_mask <- rowSums(dp > 10) > 20 # at least 20 samples have a depth of 10
# set the mask
# chrom@var.info$mask <- my_mask
mask(chrom) <- my_mask
# view the mask
# chrom@var.info$mask
mask(chrom)
```

**Comment by knausb** [comment]
Frankly, I don't use the mask anymore. These days I prefer a workflow of censoring and omitting individual variants.  I've put some info on this here:

https://knausb.github.io/vcfR_documentation/gbs_class.html

This was for a GBS data set but it generalizes well. I feel that by evaluating the individual variants I'll get greater flexibility.

**Comment by zkamvar** [comment]
I thought censoring individual variants was what the mask column was for. In the example here: https://knausb.github.io/vcfR_documentation/censoring_data.html <https://knausb.github.io/vcfR_documentation/censoring_data.html> it seems that your are destructively censoring the data. The mask allows you to get it back.


-----
Zhian N. Kamvar, Ph. D.
Postdoctoral Researcher (Everhart Lab)
Department of Plant Pathology
University of Nebraska-Lincoln

**Comment by knausb** [comment]
The mask is a vector that is the length of the number of variants. So it censors a variant over all samples. I designed it early on when I was working with data from the INFO column which also provides data summaries for each variant over all samples.  The above links show how to censor a variant for specific samples instead of all samples. I then destructively censor (omitting) either variants or samples as you've indicated. Perhaps I've become more assertive? You can censor on several criteria if you'd like. Once you're happy with your choices I like to try to reduce my memory footprint by destructively censoring. You may prefer to use the mask. I think my role here is to provide options.

**Comment by zkamvar** [comment]
Sorry. I didn't mean to be pushy; I just was looking for a bit of clarification. I was confused and interpreted "individual variants" as a specific row of the data, but you meant "individual variants" as specific elements of the data.

In any case, censoring variants on an individual basis by marking them as missing leads me to wonder if it's possible for missing data to be informative. If you have an indel where the reference has AACC and there are two alternates: TTCAGT,CGCG , if you have a sample with missing data in this region, could it possibly represent a deletion? I am genuinely curious.

**Comment by knausb** [comment]
Nope, you're not being pushy. I think you're seeing that I've changed how I work with this data through time.

I don't think I'd consider missing data as informative. Although, because the VCF specification is flexible, this may be interpreted differently be different VCF making software. To extend your example, I would think that if there was a deletion you would see the alternate reported as A or AA. An example of a microsatellite can be found in the data from the specification.

    data("vcfR_test")
    getFIX(vcfR_test)[5,]

You should see alleles of different lengths. This provides a 'positive' manner to encode indels. To score an indel as missing I would consider ambiguous and not recommended. I would think that most missing data would indicate that there is no data for that samples variant. I would like to think this could be encoded as follows.

    GT:DP ./.:0

Where GT is the genotype and DP is the depth. This would provide a positive indication that there is no genotype there because there was the read depth was zero. Although, I suppose a software could encode the same information as follows.

    GT:DP ./.

Which I would consider ambiguous. This appears to be how Tassel scores missing genotypes though.

In summary,  I think the VCF specification provides the flexibility to encode informed missingness. This leads me to concluded that missing data that you do encounter would not be worth looking into as informative.
